### PR TITLE
use a Date() based timer

### PR DESCRIPTION
**Pull request by binford2k**
Fixes half of the problem described in #304

### CODE REVIEW

File: public/js/showoff.js [hunk 1]
```diff
@@ -1190,25 +1186,29 @@ function addPreShowTips() {
...
+  var now = new Date();
+  now.setTime(now.getTime() + seconds * 1000);
+  return now;
+}
+
 function secondsToTime(sec) {
-	min = Math.floor(sec / 60)
-	sec = sec - (min * 60)
+	var min = Math.floor(sec / 60);
+	var sec = sec - (min * 60);
```
Comment (marrero984) [inline public/js/showoff.js]: Do not redeclare sec; it is a parameter

File: public/js/showoff.js [hunk 2]
```diff
@@ -1130,57 +1129,54 @@ function togglePreShow() {
...
+  $.getJSON("preshow_files", false, function(data) {
+    $('#preso').after("<div id='preshow'></div><div id='tips'></div><div id='preshow_timer'></div>");
+    $.each(data, function(i, n) {
+      if(n == "preshow.json") {
+        // has a descriptions file
+        $.getJSON("/file/_preshow/preshow.json", false, function(data) {
+          preshow_des = data;
+        })
+      } else {
+        $('#preshow').append('<img ref="' + n + '" src="/file/_preshow/' + n + '"/>');
```
Comment (marrero984) [inline public/js/showoff.js:1147]: This tries to load the README in _preshow as an img.  We can use JS to check file extension before this step or, hopefully, there is a better solution...

Comment (binford2k) [inline public/js/showoff.js:1147]: That's kind of funny. I didn't even look at the original algorithm. Honestly, all forms of image detection kind of suck here. I'm just going to ignore the README

Comment (binford2k) [inline public/js/showoff.js:1147]: Actually, no. This is a courseware problem, not a showoff problem. It's behaving as designed.

Comment (marrero984) [inline public/js/showoff.js:1147]: Okay

**Pull request merged**